### PR TITLE
Allow unquoted strings for property values, Allow collection names to have dots

### LINKED ISSUE #466: Mongo scrapbook doesn't allow non-quoted property names
The console allows this:

  db.heroes.find({ id: 2 }, { saying: 1 })
or
  db.heroes.find({ "id": 2 }, { "saying": 1 }) 

We require the second version.

### LINKED ISSUE #666: Cannot query collection under namespace - db.foo.bar.find()
Repro steps:
create collection foo.bar

Scrapbook:

db.foo.bar.find()

Action: cosmosDB.executeAllMongoCommands
Error type: Error
Error Message: Error near line 0, column 10, text '.'. Please check syntax.

Version: 0.7.1
OS: darwin

### LINKED ISSUE #717: Parse error with multiple integers in an array
db.Users.insertOne({"age": 22})
db.Users.find({"age" : { "$in" : [19, 20, 22, 25]}})    <<<<  This gives an error

### LINKED ISSUE #722: Unrecognized child type
Repro steps:

db.c.find({"a":[1,2,3]"b":1}) 

Action: vscode.workspace.onDidChangeTextDocument
Error type: AssertionError
Error Message: Unrecognized child type in parse tree.

Version: 0.7.2-alpha
OS: win32

### PR DESCRIPTION
**Pull request by PrashanthCorp**
Fixes #466 , fixes #666, fixes #722 , fixes #717

### CODE REVIEW

File: test/mongoGetCommand.test.ts [hunk 1]
```diff
@@ -107,18 +106,22 @@ suite("scrapbook parsing Tests", () => {
 
     test("second of two commands, Mac/Linux", () => {
         let line1 = "db.find()";
-        let line2 = "db.insertOne({'a':'b'})";
-        let text = `${line1}\n${line2}`;
-        let command = getCommandFromTextAtLocation(text, new Position(2, 0));
-        assert.equal(command.text, line2);
+        for (let i in [0, 1, 2]) {
+            let line2 = `db.insertOne({${wrapInQuotes("a", +i)}:'b'})`;
```
Comment (StephenWeatherford) [inline test/mongoGetCommand.test.ts]: Why the +?

File: test/mongoGetCommand.test.ts [hunk 2]
```diff
@@ -107,18 +106,22 @@ suite("scrapbook parsing Tests", () => {
...
+        }
     });
 
     test("second of two commands, Mac/Linux, semicolon", () => {
         let line1 = "db.find();";
-        let line2 = "db.insertOne({'a':'b'})";
-        let text = `${line1}\n${line2}`;
-        let command = getCommandFromTextAtLocation(text, new Position(2, 0));
-        assert.equal(command.text, line2);
+        for (let i in [0, 1, 2]) {
```
Comment (StephenWeatherford) [inline test/mongoGetCommand.test.ts]: I suggest "q" instead of "i"

File: test/mongoGetCommand.test.ts (hunk 1 above)
Comment (PrashanthCorp) [reply test/mongoGetCommand.test.ts]: I was getting an error about i being a string and not a number. The simplest way to convert (knowing that the array contains numbers) was to `+`coerce it. 

image

Comment (StephenWeatherford) [reply test/mongoGetCommand.test.ts]: for (let i: number in [0, 1, 2])
?

or just (for let i = 0; i <=2; ++i)

But not a big deal

Comment (PrashanthCorp) [reply test/mongoGetCommand.test.ts]: Typescript doesn't like the first one - no type annotation in a for ... in statement. 

I'll do the other one.

**Pull request merged**